### ISSUE TITLE
Grid Item covers items in the row above 

### ISSUE THREAD

**Issue by NinoM4ster** [swordev/suid#263]
So the way Grids work seems pretty weird. the actual element position is offset towards the top left and it uses padding to bring the content to the correct position, but that makes it so items in lower rows end up covering items above it, making their contents unclickable (plus another issue with elements below the grid).

![image](https://github.com/swordev/suid/assets/8485761/af95543c-2636-4987-aaa4-c7a73fa4175e)

Here's the playground link: https://stackblitz.com/edit/pcvhmn?file=src%2FApp.tsx

The current workaround I found (which I hate) is to set a reverse z-index on each item so that items lower down in the grid have a lower index and stand "behind" the items above it.

As far as I can tell this issue is not present in Mui: https://codesandbox.io/s/material-ui-grid-test-pqrgqj?file=/app.jsx

**Comment by juanrgm** [comment]
> As far as I can tell this issue is not present in Mui: https://codesandbox.io/s/material-ui-grid-test-pqrgqj?file=/app.jsx

You are using MUI v4.5.1 and SUID is aligned with MUI v5. 

Also, the MUI code is broken because `width`, `height` and `sx` aren't valid attributes for the `Grid` component (MUI v4).

**Comment by NinoM4ster** [comment]
> You are using MUI v4.5.1 and SUID is aligned with MUI v5.

My bad, I just forked a random sandbox link, guess it was outdated.

Anyway, I think I figured it out. The Grid Container's spacing values were really high (14), pushing the items really far apart which I apparently tried to fix by setting the Grid Item's size but that ended up breaking the UI. [Here's the fixed code](https://stackblitz.com/edit/pcvhmn-zifunz?file=src%2FApp.tsx).

Thanks for replying and sorry for not noticing it before.
